### PR TITLE
docs: Add Contour deprecation policy

### LINKED ISSUE #2749: Deprecate and then drop support for contour.heptio.com/ annotation prefix
We should eventually drop support for annotations prefixed with `contour.heptio.com/` since they've been replaced with `projectcontour.io/` for quite awhile. This could be a 2.0 thing, or we could put out a deprecation notice and then drop support a few releases later.

### LINKED ISSUE #2836: Add deprecation policy statement for `projectcontour.io` CRDs
Among the maintainers, we've always agreed that the `projectcontour.io` CRDs would abide by the Kubernetes deprecation policy's principles, but we've never written that down. 

This issue is to create a policy document so that we have it in writing, officially.

### PR DESCRIPTION
**Pull request by youngnick**
Fixes #2836 

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

File: site/_resources/deprecation-policy.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Contour Deprecation Policy
+layout: page
+---
+
+Contour published a few APIs, the most notable being the `projectcontour.io` api group of Kubernetes objects, the command line for `contour`, and the Contour configuration file.
```
Comment (stevesloka) [inline site/_resources/deprecation-policy.md:6]: nit: s/published/publishes

**Review by stevesloka — CHANGES_REQUESTED** [review]
Also, how does code in here (https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/helpers.go), for example, get deprecated / changed? It's not really part of an API version, but since public, changes can be breaking for users.

File: site/_resources/deprecation-policy.md (hunk 1 above)
Comment (youngnick) [reply site/_resources/deprecation-policy.md:6]: Fixed.

**Comment by youngnick** [comment]
@stevesloka thanks for that. I've added some statements around the helper code.

**Review by stevesloka — APPROVED** [review]
The helpers in the API sections still feel like we're going to need to version stuff all the time when those change and they should't be there (at least the one's with business logic), but otherwise lgtm.

**Comment by skriss** [comment]
> The helpers in the API sections still feel like we're going to need to version stuff all the time when those change and they should't be there (at least the one's with business logic), but otherwise lgtm.

Yeah, I hadn't really considered that when those helpers were added, but now that you bring it up, it'll certainly make me less inclined to add things in there so I don't have to maintain compatibility :)

**Review by skriss — APPROVED** [review]

**Pull request merged**